spec/requests/purchases_requests_spec.rb

```diff
@@ -264,7 +264,19 @@
 
     describe "GET #show" do
       let(:item) { create(:item) }
-      let!(:purchase) { create(:purchase, :with_items, item: item) }
+      let(:storage_location) { create(:storage_location, organization: organization, name: 'Some Storage') }
+      let(:vendor) { create(:vendor, organization: organization, business_name: 'Another Business') }
+      let!(:purchase) { create(:purchase, :with_items, comment: 'Fine day for diapers, it is.', created_at: 1.month.ago, issued_at: 1.day.ago, item: item, storage_location: storage_location, vendor: vendor) }
+
+      it "shows the purchase info" do
+        date_of_purchase = "#{1.day.ago.to_fs(:distribution_date)} (entered: #{1.month.ago.to_fs(:distribution_date)})"
```

Comment:
I think you might have to use the `freeze_time` helper here, and use `let` instead of `let!` so that the correct dates get set. Otherwise you might get flakiness here with times being off by part of a second.

Reply:
I fixed here [1364f82](https://github.com/rubyforgood/human-essentials/pull/4642/commits/1364f821be5dada65f8384c3aba9c13d580f680d)